### PR TITLE
Add support to RAG application to create a cluster for the user, or allow them to bring their own cluster

### PR DESCRIPTION
**Pull request by umeshkumhar**
- Create a cluster with appropriate nodepools/other config on behalf of the user
- Still allow bringing your own cluster
- Update marketplace blueprints with cluster creation options

### CODE REVIEW

File: applications/rag/variables.tf [hunk 1]
```diff
@@ -18,7 +18,8 @@ variable "project_id" {
 }
 
 variable "cluster_name" {
-  type = string
+  type    = string
+  default = "new-cluster"
```
Comment (imreddy13) [inline applications/rag/variables.tf:22]: why do we need this?

File: applications/rag/variables.tf [hunk 2]
```diff
@@ -28,7 +29,8 @@ variable "cluster_membership_id" {
 }
 
 variable "cluster_location" {
-  type = string
+  type    = string
+  default = "us-central1"
```
Comment (imreddy13) [inline applications/rag/variables.tf:33]: same

Comment (imreddy13) [reply applications/rag/variables.tf:33]: if we have to default, can we pick a region that has more L4 capacity ?

File: applications/rag/variables.tf [hunk 3]
```diff
@@ -79,8 +81,8 @@ variable "rag_service_account" {
 }
 
 variable "create_gcs_bucket" {
-  type = bool
-  default = false
+  type        = bool
+  default     = false
```
Comment (imreddy13) [inline applications/rag/variables.tf:85]: shouldnt it be default true? or jupyterhub/ray wont work

**Review by imreddy13 — DISMISSED** [review]

File: applications/rag/variables.tf (hunk 3 above)
Comment (umeshkumhar) [reply applications/rag/variables.tf:85]: default false is for marketplace..

**Review by imreddy13 — APPROVED** [review]

**Comment by artemvmin** [comment]
FYI something in this commit changed `tf apply` (pre-deployment) plan generation from O(seconds) to O(minutes). Unless I made a mistake, we need to solve this or revert this change.

```
# at HEAD (this commit):
time (tf apply -var-file workloads.tfvars)` 
...
real    5m16.797s
user    8m49.858s
sys     0m22.037s
```
```
# at `HEAD~1`
time (tf apply -var-file workloads.tfvars)` 
...
real    0m20.667s
user    0m26.451s
sys     0m3.219s
```

**Pull request merged**